#
# Submitted-By: Marc Kleine-Budde
# Committed-By: Robert Schwebel
#
# Error:
#
# no error
#
# Description:
#
# Link assembler softfloat functions into gcc. As the functions are
# there in that case anyway, don't add switches for float lib.
#
# State:
#
# upstream will not fix this because oabi is dead
#

Index: gcc-4.1.1/gcc/config/arm/linux-elf.h
===================================================================
--- gcc-4.1.1.orig/gcc/config/arm/linux-elf.h
+++ gcc-4.1.1/gcc/config/arm/linux-elf.h
@@ -54,7 +54,7 @@
 
 #undef  MULTILIB_DEFAULTS
 #define MULTILIB_DEFAULTS \
-	{ "marm", TARGET_ENDIAN_OPTION, "mhard-float", "mno-thumb-interwork" }
+	{ "marm", TARGET_ENDIAN_OPTION, "msoft-float" "mhard-float", "mno-thumb-interwork" }
 
 /* Now we define the strings used to build the spec file.  */
 #undef  LIB_SPEC
@@ -63,7 +63,7 @@
    %{shared:-lc} \
    %{!shared:%{profile:-lc_p}%{!profile:-lc}}"
 
-#define LIBGCC_SPEC "%{msoft-float:-lfloat} %{mfloat-abi=soft*:-lfloat} -lgcc"
+#define LIBGCC_SPEC "-lgcc"
 
 #define LINUX_TARGET_INTERPRETER "/lib/ld-linux.so.2"
 
Index: gcc-4.1.1/gcc/config/arm/t-linux
===================================================================
--- gcc-4.1.1.orig/gcc/config/arm/t-linux
+++ gcc-4.1.1/gcc/config/arm/t-linux
@@ -4,7 +4,10 @@ TARGET_LIBGCC2_CFLAGS = -fomit-frame-poi
 LIBGCC2_DEBUG_CFLAGS = -g0
 
 LIB1ASMSRC = arm/lib1funcs.asm
-LIB1ASMFUNCS = _udivsi3 _divsi3 _umodsi3 _modsi3 _dvmd_lnx
+LIB1ASMFUNCS = _udivsi3 _divsi3 _umodsi3 _modsi3 _dvmd_lnx \
+	_negdf2 _addsubdf3 _muldivdf3 _cmpdf2 _unorddf2 _fixdfsi _fixunsdfsi \
+	_truncdfsf2 _negsf2 _addsubsf3 _muldivsf3 _cmpsf2 _unordsf2 \
+	_fixsfsi _fixunssfsi _floatdidf _floatdisf
 
 # MULTILIB_OPTIONS = mhard-float/msoft-float
 # MULTILIB_DIRNAMES = hard-float soft-float
